app/src/main/java/com/mk/tasky/agenda/home/presentation/components/HomeAgendaItem.kt

```diff
@@ -0,0 +1,95 @@
+package com.mk.tasky.agenda.home.presentation.components
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Text
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.MoreHoriz
+import androidx.compose.material.icons.outlined.Circle
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.mk.tasky.ui.theme.Black
+import com.mk.tasky.ui.theme.DarkGray
+import com.mk.tasky.ui.theme.Gray
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+@Composable
+fun HomeAgendaItem(
+    title: String,
+    description: String,
+    color: Color,
+    onOptionsClick: () -> Unit,
+    startDatetime: LocalDateTime,
+    finishDateTime: LocalDateTime? = null,
+    modifier: Modifier = Modifier
+) {
+    Column(
+        modifier = modifier
+            .fillMaxWidth()
+            .heightIn(min = 130.dp)
+            .background(color = color, shape = RoundedCornerShape(20.dp))
+            .padding(16.dp)
+    ) {
+        Row(
+            modifier = Modifier.fillMaxWidth(),
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.SpaceBetween
+        ) {
+            Row(
+                verticalAlignment = Alignment.CenterVertically,
+                modifier = Modifier.weight(9f)
+            ) {
+                Icon(
+                    imageVector = Icons.Outlined.Circle,
+                    contentDescription = "title"
+                )
+                Spacer(modifier = Modifier.width(12.dp))
+                Text(
+                    text = title,
+                    fontWeight = FontWeight.Bold,
+                    fontSize = 20.sp,
+                    color = Black
+                )
+            }
+            IconButton(onClick = onOptionsClick, modifier = Modifier.weight(1f)) {
+                Icon(imageVector = Icons.Default.MoreHoriz, contentDescription = "options")
+            }
+        }
+        Text(text = description, fontSize = 14.sp, color = DarkGray)
+        val dateFormatter = DateTimeFormatter.ofPattern("dd MMM, HH:mm")
+        var itemDate = startDatetime.format(dateFormatter)
+        if (finishDateTime != null) {
+            itemDate += " - ${finishDateTime.format(dateFormatter)}"
```

Comment:
Better handle this with an effect handler since this would effectively be considered a side effect in the compose world. Not likely going to be an issue here, but just to be sure, I'd do something like this:
```
val itemDate by remember {
    derivedStateOf {
        val date = startDateTime.format(dateFormatter)
        if(finishDateTime != null) {
             date + " - ${finishDateTime.format(dateFormatter)}"
        } else date
    }
}
```

Reply:
Awesome!
Thanks for the info!